artemis-server/src/test/java/org/apache/activemq/artemis/core/config/amqpBrokerConnectivity/AMQPBrokerConnectionElementTest.java

```diff
@@ -0,0 +1,101 @@
+ * limitations under the License.
+ */
+
+package org.apache.activemq.artemis.core.config.amqpBrokerConnectivity;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+import org.junit.Test;
+
+/**
+ * Test basic API functionality of the AMQPBrokerConnectionElement type
+ */
+public class AMQPBrokerConnectionElementTest {
+
+   @Test
+   public void testEquals() {
+      AMQPBrokerConnectionElement element1 = new AMQPBrokerConnectionElement();
+      AMQPBrokerConnectionElement element2 = new AMQPBrokerConnectionElement();
+
+      assertEquals(element1, element2);
+
+      // Name
+      element1.setName("test");
+      assertNotEquals(element1, element2);
+      element2.setName("test");
+      assertEquals(element1, element2);
+
+      // Match Address
+      element1.setMatchAddress("test");
+      assertNotEquals(element1, element2);
+      element2.setMatchAddress("test");
+      assertEquals(element1, element2);
+
+      // Queue Name
+      element1.setQueueName("test");
+      assertNotEquals(element1, element2);
+      element2.setQueueName("test");
+      assertEquals(element1, element2);
+
+      // Type
+      element1.setType(AMQPBrokerConnectionAddressType.MIRROR);
+      assertNotEquals(element1, element2);
+      element2.setType(AMQPBrokerConnectionAddressType.MIRROR);
+      assertEquals(element1, element2);
+   }
+
+   @Test
+   public void testHashCode() {
+      AMQPBrokerConnectionElement element1 = new AMQPBrokerConnectionElement();
+      AMQPBrokerConnectionElement element2 = new AMQPBrokerConnectionElement();
+
+      assertEquals(element1.hashCode(), element2.hashCode());
+
+      // Name
+      element1.setName("test");
+      assertNotEquals(element1.hashCode(), element2.hashCode());
+      element2.setName("test");
+      assertEquals(element1.hashCode(), element2.hashCode());
+
+      // Match Address
+      element1.setMatchAddress("test");
+      assertNotEquals(element1.hashCode(), element2.hashCode());
+      element2.setMatchAddress("test");
+      assertEquals(element1.hashCode(), element2.hashCode());
+
+      // Queue Name
+      element1.setQueueName("test");
+      assertNotEquals(element1.hashCode(), element2.hashCode());
+      element2.setQueueName("test");
+      assertEquals(element1.hashCode(), element2.hashCode());
+
+      // Type
+      element1.setType(AMQPBrokerConnectionAddressType.MIRROR);
+      assertNotEquals(element1.hashCode(), element2.hashCode());
+      element2.setType(AMQPBrokerConnectionAddressType.MIRROR);
+      assertEquals(element1.hashCode(), element2.hashCode());
+
+      // Parent
+      final AMQPBrokerConnectConfiguration parent = new AMQPBrokerConnectConfiguration();
+
+      element1.setParent(parent);
+      assertNotEquals(element1.hashCode(), element2.hashCode());
+      element2.setParent(parent);
+      assertEquals(element1.hashCode(), element2.hashCode());
```

Comment:
Specifically, also checking equals() here would presumably show it violating the joint contract (the 2 beeing equal throughput but having different, and changing, hashCodes).